rand_core/src/lib.rs

```diff
@@ -185,22 +207,20 @@ pub trait SeedableRng<Seed>: Rng {
     fn from_seed(seed: Seed) -> Self;
 }
 
-
-/// Error type for cryptographic generators. Technically external generators
-/// such as the operating system or hardware generators could fail. A PRNG
-/// (algorithm) could also fail if it detects cycles, though most PRNGs have
-/// sufficiently long cycles that looping is not usually feasible.
-/// 
-/// TODO: how should error details be reported?
-#[derive(Debug)]
-pub struct Error;
-
-#[cfg(feature="std")]
-impl From<::std::io::Error> for Error {
```

Comment:
I did not add this `From` implementation yet.
I actually had one, that would map the IO errors you could get for `OsRng` to the appropriate `RngError`.
But I am not sure it would always be correct for other RNGs implementations. It are the more exotic ones that produce errors...

Not adding one force RNG implementations to put a little more thought into their error handling. Which might be a plus... For `ReadRng` the missing implementation did not really make it more difficult, and I also expect it not to for `OsRng`.

Reply:
This sounds sensible to me.